torch/nn/modules/instancenorm.py

```diff
@@ -11,22 +13,27 @@ def __init__(self, num_features, eps=1e-5, momentum=0.1, affine=False,
     def _check_input_dim(self, input):
         return NotImplemented
 
-    def _load_state_dict_key_mismatch(self, full_name, name, is_missing):
-        if not is_missing and not self.track_running_stats and \
-                name in ('running_mean', 'running_var'):
-            raise KeyError(
-                'Unexpected running stats buffer "{name}" in state_dict for '
-                '{klass} with track_running_stats=False. If you are trying to '
-                'load a checkpoint saved before 0.4.0, this may be expected '
-                'because {klass} does not track running stats by default '
-                'anymore since 0.4.0. If running stats are not used, remove '
-                'them from state_dict before calling load_state_dict. '
-                'Otherwise, set track_running_stats=True in {klass} to use '
-                'running stats. See the documentation of {klass} for more '
-                'details.'
-                .format(name=full_name, klass=self.__class__.__name__))
-        super(_InstanceNorm, self)._load_state_dict_key_mismatch(
-            full_name, name, is_missing)
+    def load_local_state_dict(self, local_state_dict, version, strict):
+        # at version 0: removed running_mean and running_var when
+        # track_running_stats=False (default)
+        if version is None and not self.track_running_stats:
+            running_stats_keys = tuple(k for k in ('running_mean', 'running_var') if k in local_state_dict)
+            if len(running_stats_keys) > 0:
+                warnings.warn(
+                    'Unexpected running stats buffer(s) {names} in state_dict '
+                    'for {klass} with track_running_stats=False. If you are '
+                    'trying to load a checkpoint saved before 0.4.0, this may '
+                    'be expected because {klass} does not track running stats '
+                    'by default anymore since 0.4.0. These buffers won\'t be '
+                    'actually loaded. If these stats are actually needed, set '
+                    'track_running_stats=True in {klass} to enable running '
+                    'stats. See the documentation of {klass} for details.'
+                    .format(names=" and ".join('"{}"'.format(k) for k in running_stats_keys),
+                            klass=self.__class__.__name__))
```

Comment:
Why is that a warning now? It should really be an error, and someone should either patch their state dict, or change the flag to the constructor of this module

Reply:
The more I worked on tutorials code (and my research code), the more I want this to be a warning. So many people will try using community repo code to load existing checkpoints trained with InstanceNorm, and we can't rely on the repos being updated. From the users perspective, they may not be familiar with how state_dict works, or even not very familiar with PyTorch, but just want to run a cool model like neural style transfer. So I really really want to have this as warning.

Reply:
I'm a bit worried about this. It is completely inconsistent with our `strict` setting, which means it basically violates the contract of `load_state_dict`. Will we always want to make those things warnings?

Your argument make sense, but this can really be fixed extremely easily (`strict=False`), so it's not that convincing for me.

Reply:
The original IN behavior is really subtle. It only uses running stats when `.train(False)`, but not `.eval()`. And using running stats is a non-standard thing to do with IN.

 So I think among people that loads a `state_dict` with `running_*`, ones who actually wants to use them are extremely few. Since we don't want to advocate using `strict=False` as it suppresses other mismatches, I think it is reasonable to break the strict protocol here a bit, and print the warning.